### DISCUSSION TITLE
Bonsai does not recognize "Com "X" " when trying to recieve data from the Arduino and send data at the same time

### DISCUSSION THREAD

**Discussion by Mardornack** [bonsai-rx/bonsai#1351] [Q&A]
Hope you all are having a great day!
I am currently having trouble with my Bonsai-Arduino match.
Bonsai is reading a TTL that is sended via Arduino´s USB port cable "COM". But in my desire script there is also a command that allows me to trigger the Arduino code, so that the signal (TTL) begins exactly when i want. Apparently the problem is that the script/bonsai/something does not support the same port for reciving and giving information. Thanks a lot!
Here is the code:
![Registro](https://github.com/bonsai-rx/bonsai/assets/128514957/906e38e2-d6e6-45d6-adce-55615ec21c5e)

**Comment by bruno-f-cruz** [comment]
Can you add the code of the workflow? Thanks

**Reply by Mardornack** [reply]
<?xml version="1.0" encoding="utf-8"?>
<WorkflowBuilder Version="2.7.1"
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                 xmlns:gl="clr-namespace:Bonsai.Shaders;assembly=Bonsai.Shaders"
                 xmlns:gli="clr-namespace:Bonsai.Shaders.Input;assembly=Bonsai.Shaders"
                 xmlns:rx="clr-namespace:Bonsai.Reactive;assembly=Bonsai.Core"
                 xmlns:ard="clr-namespace:Bonsai.Arduino;assembly=Bonsai.Arduino"
                 xmlns:io="clr-namespace:Bonsai.IO;assembly=Bonsai.System"
                 xmlns:wie="clr-namespace:Bonsai.Windows.Input;assembly=Bonsai.Windows.Input"
                 xmlns="https://bonsai-rx.org/2018/workflow">
  <Workflow>
    <Nodes>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="gl:UpdateFrame" />
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="gli:Mouse">
          <gli:Index>0</gli:Index>
        </Combinator>
      </Expression>
      <Expression xsi:type="MemberSelector">
        <Selector>X</Selector>
      </Expression>
      <Expression xsi:type="MemberSelector">
        <Selector>Y</Selector>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="rx:Zip" />
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="gl:UpdateFrame" />
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="gli:Mouse">
          <gli:Index>1</gli:Index>
        </Combinator>
      </Expression>
      <Expression xsi:type="MemberSelector">
        <Selector>X</Selector>
      </Expression>
      <Expression xsi:type="MemberSelector">
        <Selector>Y</Selector>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="rx:Zip" />
      </Expression>
      <Expression xsi:type="Disable">
        <Builder xsi:type="Combinator">
          <Combinator xsi:type="rx:Timer">
            <rx:DueTime>PT0S</rx:DueTime>
            <rx:Period>PT0.02S</rx:Period>
          </Combinator>
        </Builder>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="ard:DigitalInput">
          <ard:PortName>COM6</ard:PortName>
          <ard:Pin>13</ard:Pin>
        </Combinator>
      </Expression>
      <Expression xsi:type="Disable">
        <Builder xsi:type="Parse">
          <Pattern>%B</Pattern>
        </Builder>
      </Expression>
      <Expression xsi:type="Disable">
        <Builder xsi:type="Combinator">
          <Combinator xsi:type="BooleanProperty">
            <Value>true</Value>
          </Combinator>
        </Builder>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="gl:UpdateFrame" />
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="rx:Timestamp" />
      </Expression>
      <Expression xsi:type="MemberSelector">
        <Selector>Timestamp.Millisecond,Value.TimeStep.ElapsedTime,Timestamp.Second</Selector>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="rx:Zip" />
      </Expression>
      <Expression xsi:type="Disable">
        <Builder xsi:type="Combinator">
          <Combinator xsi:type="rx:SampleInterval">
            <rx:Interval>PT0.05S</rx:Interval>
          </Combinator>
        </Builder>
      </Expression>
      <Expression xsi:type="Disable">
        <Builder xsi:type="io:CsvWriter">
          <io:FileName>C:\Users\barney\Desktop\Bonsai\Registro Prueba\Prueba.csv</io:FileName>
          <io:Append>false</io:Append>
          <io:Overwrite>true</io:Overwrite>
          <io:Suffix>None</io:Suffix>
          <io:IncludeHeader>false</io:IncludeHeader>
        </Builder>
      </Expression>
      <Expression xsi:type="Disable">
        <Builder xsi:type="Combinator">
          <Combinator xsi:type="rx:WithLatestFrom" />
        </Builder>
      </Expression>
      <Expression xsi:type="io:CsvWriter">
        <io:FileName>C:\Users\barney\Desktop\Bonsai\Registro Prueba\20230502_00.csv</io:FileName>
        <io:Append>false</io:Append>
        <io:Overwrite>true</io:Overwrite>
        <io:Suffix>None</io:Suffix>
        <io:IncludeHeader>false</io:IncludeHeader>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="gl:CreateWindow">
          <gl:Width>640</gl:Width>
          <gl:Height>480</gl:Height>
          <gl:VSync>On</gl:VSync>
          <gl:SwapSync>true</gl:SwapSync>
          <gl:ClearColor>BlanchedAlmond</gl:ClearColor>
          <gl:ClearMask>DepthBufferBit ColorBufferBit</gl:ClearMask>
          <gl:CursorVisible>true</gl:CursorVisible>
          <gl:Location xsi:nil="true" />
          <gl:WindowBorder>Resizable</gl:WindowBorder>
          <gl:WindowState>Normal</gl:WindowState>
          <gl:DisplayDevice>Primary</gl:DisplayDevice>
          <gl:TargetRenderFrequency>50</gl:TargetRenderFrequency>
          <gl:TargetUpdateFrequency xsi:nil="true" />
          <gl:RenderState />
          <gl:GraphicsMode>
            <gl:ColorFormat>
              <gl:Red>8</gl:Red>
              <gl:Green>8</gl:Green>
              <gl:Blue>8</gl:Blue>
              <gl:Alpha>8</gl:Alpha>
            </gl:ColorFormat>
            <gl:Depth>16</gl:Depth>
            <gl:Stencil>0</gl:Stencil>
            <gl:Samples>0</gl:Samples>
            <gl:AccumulatorFormat>
              <gl:Red>0</gl:Red>
              <gl:Green>0</gl:Green>
              <gl:Blue>0</gl:Blue>
              <gl:Alpha>0</gl:Alpha>
            </gl:AccumulatorFormat>
            <gl:Buffers>2</gl:Buffers>
            <gl:Stereo>false</gl:Stereo>
          </gl:GraphicsMode>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="io:CreateSerialPort">
          <io:Name>myArduino</io:Name>
          <io:PortName>COM3</io:PortName>
          <io:BaudRate>57600</io:BaudRate>
          <io:Parity>None</io:Parity>
          <io:ParityReplace>63</io:ParityReplace>
          <io:DataBits>8</io:DataBits>
          <io:StopBits>One</io:StopBits>
          <io:Handshake>None</io:Handshake>
          <io:DiscardNull>false</io:DiscardNull>
          <io:DtrEnable>false</io:DtrEnable>
          <io:RtsEnable>false</io:RtsEnable>
          <io:ReadBufferSize>4096</io:ReadBufferSize>
          <io:WriteBufferSize>2048</io:WriteBufferSize>
          <io:ReceivedBytesThreshold>1</io:ReceivedBytesThreshold>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="wie:KeyDown">
          <wie:Filter>A</wie:Filter>
          <wie:SuppressRepetitions>true</wie:SuppressRepetitions>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="StringProperty">
          <Value>a</Value>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="io:SerialStringWrite">
          <io:PortName>myArduino</io:PortName>
          <io:NewLine>\r\n</io:NewLine>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="wie:KeyDown">
          <wie:Filter>S</wie:Filter>
          <wie:SuppressRepetitions>true</wie:SuppressRepetitions>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="StringProperty">
          <Value>s</Value>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="io:SerialStringWrite">
          <io:PortName>myArduino</io:PortName>
          <io:NewLine>\r\n</io:NewLine>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="io:SerialStringRead">
          <io:PortName>myArduino</io:PortName>
          <io:NewLine>\r\n</io:NewLine>
        </Combinator>
      </Expression>
      <Expression xsi:type="Parse">
        <Pattern>%s</Pattern>
        <Separator />
      </Expression>
    </Nodes>
    <Edges>
      <Edge From="0" To="1" Label="Source1" />
      <Edge From="1" To="2" Label="Source1" />
      <Edge From="1" To="3" Label="Source1" />
      <Edge From="2" To="4" Label="Source1" />
      <Edge From="3" To="4" Label="Source2" />
      <Edge From="4" To="17" Label="Source2" />
      <Edge From="5" To="6" Label="Source1" />
      <Edge From="6" To="7" Label="Source1" />
      <Edge From="6" To="8" Label="Source1" />
      <Edge From="7" To="9" Label="Source1" />
      <Edge From="8" To="9" Label="Source2" />
      <Edge From="9" To="17" Label="Source1" />
      <Edge From="10" To="20" Label="Source1" />
      <Edge From="11" To="12" Label="Source1" />
      <Edge From="11" To="17" Label="Source4" />
      <Edge From="12" To="13" Label="Source1" />
      <Edge From="14" To="15" Label="Source1" />
      <Edge From="15" To="16" Label="Source1" />
      <Edge From="16" To="17" Label="Source3" />
      <Edge From="17" To="18" Label="Source1" />
      <Edge From="17" To="20" Label="Source2" />
      <Edge From="18" To="19" Label="Source1" />
      <Edge From="20" To="21" Label="Source1" />
      <Edge From="24" To="25" Label="Source1" />
      <Edge From="25" To="26" Label="Source1" />
      <Edge From="27" To="28" Label="Source1" />
      <Edge From="28" To="29" Label="Source1" />
      <Edge From="30" To="31" Label="Source1" />
    </Edges>
  </Workflow>
</WorkflowBuilder>

**Reply by bruno-f-cruz** [reply]
Can you zip the workflow as indicated in https://github.com/orgs/bonsai-rx/discussions/864

Thanks
